test_scripts/Resumption/Handling_errors_from_HMI/022_resume_all_data_with_one_error_rpc_unexpected_disconnect_background.lua

```diff
@@ -0,0 +1,80 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md
+--
+-- Requirement summary:TBD
+--
+-- Description:
+-- In case:
+-- 1. App is in Background HMI level
+-- 2. Rpc_n for resumption is added by app
+-- 3. Unexpected disconnect and reconnect are performed
+-- 4. App reregisters with actual HashId
+-- 5. Rpc_n request is sent from SDL to HMI during resumption
+-- 6. HMI responds with error resultCode to Rpc_n request
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+-- 3. not restore Background HMI level
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.application1.registerAppInterfaceParams.isMediaApplication = false
+config.application1.registerAppInterfaceParams.appHMIType = { "DEFAULT" }
+
+--[[ Common Functions ]]
+local function deactivateAppToBackground()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnAppDeactivated", {
+      appID = common.getHMIAppId()
+    })
+  common.getMobileSession():ExpectNotification("OnHMIStatus",
+    {hmiLevel = "BACKGROUND", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+end
+
+local function absenceResumptionToBacground()
+  common.getHMIConnection():ExpectRequest("BasicCommunication.OnResumeAudioSource")
+  :Times(0)
+
+  common.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp")
+  :Times(0)
+
+  common.getMobileSession():ExpectNotification("OnHMIStatus",
+    { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+end
+
+local function activateApp()
+  local requestId = common.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = common.getHMIAppId() })
+  common.getHMIConnection():ExpectResponse(requestId)
+  common.getMobileSession():ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+end
+
```

Comment:
@GetmanetsIrina Propose to move `deactivateAppToBackground()` and `activateApp()` to `common`